assets/ca/ca.querytranslator.js

```diff
@@ -0,0 +1,387 @@
+/* ----------------------------------------------------------------------
+ * js/ca/ca.querytranslator.js
+ * ----------------------------------------------------------------------
+ * CollectiveAccess
+ * Open-source collections management software
+ * ----------------------------------------------------------------------
+ *
+ * Software by Whirl-i-Gig (http://www.whirl-i-gig.com)
+ * Copyright 2014 Whirl-i-Gig
+ *
+ * For more information visit http://www.CollectiveAccess.org
+ *
+ * This program is free software; you may redistribute it and/or modify it under
+ * the terms of the provided license as published by Whirl-i-Gig
+ *
+ * CollectiveAccess is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTIES whatsoever, including any implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * This source code is free and modifiable under the terms of
+ * GNU General Public License. (http://www.gnu.org/copyleft/gpl.html). See
+ * the "license.txt" file for details, or visit the CollectiveAccess web site at
+ * http://www.CollectiveAccess.org
+ *
+ * ----------------------------------------------------------------------
+ */
+
+var caUI = caUI || {};
+
+/**
+ * This file defines two functions in the `caUI` namespace, for converting between (one function for each direction of
+ * conversion) the CA search query syntax and the jQuery query builder hierarchical rule set structure.
+ *
+ * The only range filter that is supported here is `between`.  The Lucene / CA search syntax does not readily support
+ * `greater_than` or `less_than`.
+ */
+(function () {
+	var escapeValue, getTokenList, shiftToken, tokensToRuleSet,
```

Comment:
http://www.comicsandmemes.com/wp-content/uploads/wonder-woman-Oh-no-you-didnt-meme.jpg